repose-aggregator/src/docs/asciidoc/services/datastores.adoc

```diff
@@ -3,4 +3,293 @@
+<distributed-datastore xmlns='http://docs.openrepose.org/repose/distributed-datastore/v1.0'>
+    <allowed-hosts allow-all="false"/>
+    <port-config>
+        <port port="9999" cluster="repose"/>
+        <port port="7777" cluster="repose" node="node2"/>
+    </port-config>
+</distributed-datastore>
+----
+
+== Remote Datastore
+=== Configuration
+* *Default Configuration:* remote-datastore.cfg.xml
+* *Released:* v8.5.1.0
+* link:../schemas/remote-datastore.xsd[Schema]
+
+At times, Repose instances may need to share information between nodes that are unaware of each other.
+An example of this is a dynamic containerized environment like OpenShift or other 12-Factor environment.
+
+The Remote Datastore Service allows dynamic isolated Repose instances to use a single static Repose instance's object store.
+The Remote Datastore Service communicates to the configure host through the configured port.
+If the Remote Datastore Service is unable to communicate with the configured object store, it will fall back on the local datastore temporarily.
+The Remote Datastore Service will return to using the configured object store as soon as it becomes reachable again.
+
+The static Repose instance is simply configured as a single node cluster with the distributed datastore service enabled.
+This distributed datastore is typically configured with both a `keystore` so that HTTPS is used and a `truststore` so that all clients must properly authenticate before the session is established.
+Then all of the dynamic clients are configured with a remote datastore and the same `keystore` to authenticate with and the same `truststore` to confirm the HTTPS session.
+
+The following is a basic sample configuration.
+[source,xml]
+.remote-datastore.cfg.xml
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<remote-datastore>
+    <cluster id="repose"
+             host="remote.example.com"
+             port="8080"/>
+</remote-datastore>
+----
+
+==== Full Configuration
+
+[source,xml]
+.remote-datastore.cfg.xml
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<remote-datastore>
+    <cluster id="repose"                             <!--1-->
+             host="remote.example.com"               <!--2-->
+             port="8080"                             <!--3-->
+             useSSL="true"                           <!--4-->
+             connection-pool-id="remote-datastore"/> <!--5-->
+</remote-datastore>
+----
+<1> Defines the id of the cluster this configuration is for.
+<2> Defines the host providing the remote datastore.
+<3> Defines the port on which the remote datastore is listening (1-65535).
+<4> Defines if the remote datastore is expecting a secure connection request (Default: True).
+<5> Http Connection Pool ID to use when connecting to the Remote Datastore.
+
+NOTE: Refer to the <<http-connection-pool.adoc#,http-connection-pool.cfg.xml>> documentation for setting up Client Authentication on the connection pool used to connect to the Remote Datastore.
```

Comment:
Oh man, you're right! I distinctly remember talking about this, and I was convinced I had added it to the style guide. Bad job, self.